src/components/gltf-model-plus.js

```diff
@@ -757,18 +772,9 @@ export async function loadGLTF(src, contentType, onProgress, jsonPreprocessor) {
             disposables.add(obj.geometry);
           }
 
-          if (obj.material) {
-            if (obj.material.map) disposables.add(obj.material.map);
-            if (obj.material.lightMap) disposables.add(obj.material.lightMap);
-            if (obj.material.bumpMap) disposables.add(obj.material.bumpMap);
-            if (obj.material.normalMap) disposables.add(obj.material.normalMap);
-            if (obj.material.specularMap) disposables.add(obj.material.specularMap);
-            if (obj.material.envMap) disposables.add(obj.material.envMap);
-            if (obj.material.aoMap) disposables.add(obj.material.aoMap);
-            if (obj.material.metalnessMap) disposables.add(obj.material.metalnessMap);
-            if (obj.material.roughnessMap) disposables.add(obj.material.roughnessMap);
-            if (obj.material.emissiveMap) disposables.add(obj.material.emissiveMap);
-          }
+          mapMaterials(obj, function (m) {
+            disposables.add(m);
+          });
```

Comment:
smart

Reply:
Yeah, in making sure we were calling `disposeMaterial` here it became clear that we didn't need to be so verbose. Also it was actually a bug before because we weren't calling dispose on the material itself.